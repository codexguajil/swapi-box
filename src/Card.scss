.card {
    border: 2px solid hsl(0, 0%, 0%);
    font-size: 16px;
    height: 250px;
    margin: 20px;
    width: 200px;
}

.residents {
    height: 25px;
    margin: 0;
    overflow: scroll;
}

.resident {
    margin-block-start: 0;
}