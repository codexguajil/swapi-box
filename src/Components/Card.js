import React, { Component } from 'react';

export default class Card extends Component {

    render() {
      return (
        <div className="card">
          <h1>{this.props.data.name}</h1>
          <h3>{this.props.data.homeworld}</h3>
          <h4>{this.props.data.species}</h4>
          <h4>{this.props.data.population}</h4>
        </div>
      )
    }
  }