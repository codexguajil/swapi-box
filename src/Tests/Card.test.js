import React from 'react';
import Card from '../Components/Card';
import { shallow } from 'enzyme';

describe("Card", () => {
    let mockProps = {
        name: 'edgar',
        Homeworld: 'earth',
        occupation: 'student'
    }

    let wrapper = <Card data={mockProps} />

    it("should render a card with the correct passed in props", () => {
        expect(wrapper).toMatchSnapshot();
    })
})