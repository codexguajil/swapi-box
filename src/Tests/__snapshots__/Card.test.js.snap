// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a card with the correct passed in props 1`] = `
<Card
  data={
    Object {
      "Homeworld": "earth",
      "name": "edgar",
      "occupation": "student",
    }
  }
/>
`;
